crates/octez/src/async/bootstrap.rs

```diff
@@ -5,57 +5,88 @@ use serde::de::{Error, Visitor};
 pub struct BootstrapAccount {
     public_key: PublicKey,
     amount_mutez: u64,
 }
 
-impl TryFrom<&Value> for BootstrapAccount {
-    type Error = anyhow::Error;
+impl Serialize for BootstrapAccount {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut s = serializer.serialize_seq(Some(2))?;
+        s.serialize_element(&self.public_key.to_string())?;
+        s.serialize_element(&self.amount_mutez.to_string())?;
+        s.end()
+    }
+}
 
-    fn try_from(value: &Value) -> Result<Self, Self::Error> {
-        let array = value
-            .as_array()
-            .ok_or(anyhow::anyhow!("value is not a valid json array"))?;
-        if array.len() != 2 {
-            return Err(anyhow::anyhow!(
-                "value is not in the acceptable format for bootstrap accounts"
-            ));
+struct BootstrapAccountVisitor;
+
+impl<'de> Visitor<'de> for BootstrapAccountVisitor {
+    type Value = BootstrapAccount;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a list representing a bootstrap account")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'de>,
+    {
+        if let Some(v) = seq.size_hint() {
+            if v != 2 {
+                return Err(A::Error::custom(
+                    "the input list should contain exactly two elements: address public key and account balance as string",
+                ));
+            }
+        }
+
+        let mut first_element: Option<String> = None;
+        let mut second_element: Option<String> = None;
+        while let Some(current) = seq.next_element()? {
+            if first_element.is_none() {
+                first_element.replace(current);
+            } else if second_element.is_none() {
+                second_element.replace(current);
+            } else {
+                return Err(A::Error::custom(
+                    "the input list should only contain two elements",
+                ));
+            }
         }
-        let public_key = array
-            .first()
-            .unwrap()
-            .as_str()
-            .ok_or(anyhow::anyhow!("'public_key' is not a valid string"))?;
-        let amount_mutez = array
-            .get(1)
-            .unwrap()
-            .as_str()
-            .ok_or(anyhow::anyhow!(
-                "'amount' is not a valid string representing an unsigned integer"
-            ))?
-            .parse::<u64>();
-        if amount_mutez.is_err() {
-            return Err(anyhow::anyhow!(
-                "'amount' is not a valid string representing an unsigned integer"
+
+        if first_element.is_none() || second_element.is_none() {
+            return Err(A::Error::custom(
+                "the input list should contain exactly two elements: address public key and account balance as string",
```

Comment:
nit: Slightly cleaner if the repeated error messages were moved into their own variable and reused